src/mito2/src/compaction/twcs.rs

```diff
@@ -84,37 +77,90 @@ impl TwcsPicker {
 
             // we only remove deletion markers once no file in current window overlaps with any other window.
             let found_runs = sorted_runs.len();
-            let filter_deleted = !files.overlapping && (found_runs == 1 || max_runs == 1);
+            let mut filter_deleted = !files.overlapping && (found_runs == 1 || max_runs == 1);
```

Comment:
Consider assigning this flag in one line.
```suggestion
            let filter_deleted = !files.overlapping && (found_runs == 1 || max_runs == 1) && !self.append_mode;
```

Reply:
> Consider assigning this flag in one line.

fixed in 2aa6d4263a